recipes/sdl/3.x/conanfile.py

```diff
@@ -0,0 +1,227 @@
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    @property
+    def _is_unix_sys(self):
+        """ True for UNIX but not Macos/Android"""
+        # CMakeLists.txt#L110
+        return self.settings.os in ("Linux", "FreeBSD")
+
+    @property
+    def _needs_libusb(self):
+        # CMakeLists.txt#L134
+        # TODO: Add option for libusb
+        return (self.options.get_safe("hidapi") and
+                (not is_apple_os(self) or self.settings.os == "Macos") and
+                self.settings.os != "Android")
+
+    @property
+    def _supports_opengl(self):
+        # CMakeLists.txt#L297
+        # TODO: Add option for opengl
+        return (self.options.get_safe("video")
+                and self.settings.os not in ("iOS", "tvOS", "watchOS"))
+
+    @property
+    def _supports_libudev(self):
+        # CMakeLists.txt#L351&L1618
+        # TODO: Add option for libudev
+        return self.settings.os in ("Linux", "FreeBSD")
+
+    @property
+    def _supports_pulseaudio(self):
+        # CMakeLists.txt#L372
+        # TODO: Add option for pulseaudio
+        return self._is_unix_sys and self.options.get_safe("audio")
+
+    @property
+    def _supports_dbus(self):
+        # CMakeLists.txt#292
+        # TODO: Add option for dbus
+        return self._is_unix_sys
+
+    def requirements(self):
+        if self._needs_libusb:
+            self.requires("libusb/1.0.26")
+        if self._supports_opengl:
+            self.requires("opengl/system")
+        if self._supports_libudev:
+            self.requires("libudev/system")
+        if self._supports_dbus:
+            self.requires("dbus/1.15.8")
+        if self.options.get_safe("pulseaudio"):
+            self.requires("pulseaudio/17.0")
+        if self.options.get_safe("alsa"):
+            self.requires("libalsa/1.2.12")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.24 <4]")
+        if self.settings.os == "Linux" and not self.conf.get("tools.gnu:pkg_config", check_type=str):
+            self.tool_requires("pkgconf/[>=2.2 <3]")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["SDL_SHARED"] = self.options.shared
+        tc.cache_variables["SDL_STATIC"] = not self.options.shared
+        # Todo: Remove after recipe develop is finished
+        tc.cache_variables["SDL_TEST_LIBRARY"] = True
+        tc.cache_variables["SDL_TESTS"] = True
+        tc.cache_variables["SDL_EXAMPLES"] = True
+        tc.cache_variables["CMAKE_TRY_COMPILE_CONFIGURATION"] = str(self.settings.build_type)
+        if self._needs_libusb:
+            tc.cache_variables["SDL_HIDAPI_LIBUSB"] = True
+            tc.cache_variables["SDL_HIDAPI_LIBUSB_SHARED"] = self.dependencies["libusb"].options.get_safe("shared", False)
+        if self.options.get("pulseaudio"):
+            tc.cache_variables["SDL_PULSEAUDIO"] = True
```

Comment:
Is there a reason you're not assigning `bool(self.options.get("pulseaudio"))` h

Reply:
We still need the conditional for the variable below, as accessing a non existing dependencie errors out, so at that point I think this way is cleaner for both :)